test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -237,25 +449,114 @@ contract LoansForkTest is LoansTestBase {
         assertGt(protocolFee, 0);
     }
 
-    function closeAndCheckLoan(uint loanId, uint loanAmount) internal {
+    function closeAndCheckLoan(uint loanId, uint loanAmount, uint lateFee) internal {
         // Track balances before closing
         uint userCashBefore = pair.cashAsset.balanceOf(user);
         uint userUnderlyingBefore = pair.underlying.balanceOf(user);
 
-        // no price change so underlying out should be underlying in minus slippage
+        // no price change so underlying out should be underlying in minus slippage and minus late fees
         uint minUnderlyingOut = underlyingAmount;
-        uint minUnderlyingOutWithSlippage = minUnderlyingOut * (100 - slippage) / 100;
+        uint minUnderlyingOutWithSlippage = (minUnderlyingOut * (100 - slippage) / 100) - lateFee;
         uint underlyingOut = closeLoan(pair, user, loanId, minUnderlyingOutWithSlippage);
         assertGe(underlyingOut, minUnderlyingOutWithSlippage);
         // Verify balance changes
         assertEq(userCashBefore - pair.cashAsset.balanceOf(user), loanAmount);
         assertEq(pair.underlying.balanceOf(user) - userUnderlyingBefore, underlyingOut);
     }
 
+    function createEscrowOffer(uint duration) internal returns (uint offerId) {
+        vm.startPrank(escrowSupplier);
+        pair.underlying.approve(address(pair.escrowNFT), underlyingAmount);
+        offerId = pair.escrowNFT.createOffer(
+            underlyingAmount,
+            duration,
+            interestAPR,
+            maxGracePeriod,
+            lateFeeAPR,
+            0 // minEscrow
+        );
+        vm.stopPrank();
+    }
+
+    function openEscrowLoan(uint minLoanAmount, uint providerOfferId, uint escrowOfferId, uint escrowFee)
+        internal
+        returns (uint loanId, uint providerId, uint loanAmount)
+    {
+        vm.startPrank(user);
+        // Approve underlying amount plus escrow fee
+        pair.underlying.approve(address(pair.loansContract), underlyingAmount + escrowFee);
+
+        (loanId, providerId, loanAmount) = pair.loansContract.openEscrowLoan(
+            underlyingAmount,
+            minLoanAmount,
+            ILoansNFT.SwapParams(0, address(pair.loansContract.defaultSwapper()), ""),
+            ILoansNFT.ProviderOffer(pair.providerNFT, providerOfferId),
+            ILoansNFT.EscrowOffer(pair.escrowNFT, escrowOfferId),
+            escrowFee
+        );
+        vm.stopPrank();
+    }
+
+    function _createEscrowOffers() internal returns (uint offerId, uint escrowOfferId) {
```

Comment:
is there any reasoning on what are the `_` prefixed internal methods, and the non-prefixed?